Name: Advanced Reset Options - DE
Version: 2.2.0-1
Author: DiplPhy
Description: Advanced reset options patch for german Pre3 devices. Based on the patch of Jason Robitaille. Adds restart luna, restart device, restart java to Powerup Menu and modifies shutdown to stutdown instantly.

--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -3698,7 +3698,7 @@
 		    var f = function(stageController){
 		      stageController.pushScene('poweroffalert',this);
 		    }.bind(this);
-		    var params = {name: 'PowerOffAlert', height: 175,lightweight:true};		    	    
+		    var params = {name: 'PowerOffAlert', height: 290,lightweight:true};		    	    
 		    appController.createStageWithCallback(params,f,'popupalert');
         }
 	//Close the Storage Alert and create a USB dashboard.
--- .orig/usr/lib/luna/system/luna-systemui/resources/de/views/poweroffalert/poweroffalert-scene.html
+++ /usr/lib/luna/system/luna-systemui/resources/de/views/poweroffalert/poweroffalert-scene.html
@@ -1,5 +1,7 @@
 <div class="system-notification power-off">
 		<div class="palm-notification-button affirmative" x-mojo-tap-highlight="immediate" id='flightmode'></div>
-		<div class="palm-notification-button negative" x-mojo-tap-highlight="immediate" id='poweroff'>Ausschalten</div>
+		<div class="palm-notification-button alternate" x-mojo-tap-highlight="immediate" id='luna' x-mojo-loc=''>Luna Neustart</div>
+		<div class="palm-notification-button alternate" x-mojo-tap-highlight="immediate" id='restart' x-mojo-loc=''>Gerät Neustart</div>
+		<div class="palm-notification-button negative" x-mojo-tap-highlight="immediate" id='shutdown' x-mojo-loc=''>Ausschalten</div>
 		<div class="palm-notification-button" x-mojo-tap-highlight="immediate" id='cancel'>Abbrechen</div>
 </div>
--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/poweroffalert-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/poweroffalert-assistant.js
@@ -7,7 +7,9 @@
     
     setup: function() {		
 		var flightMode = this.flightMode.bindAsEventListener(this);
-		var powerOff = this.powerOff.bindAsEventListener(this);
+		var luna = this.luna.bindAsEventListener(this);
+		var restart = this.restart.bindAsEventListener(this);
+		var shutdown = this.shutdown.bindAsEventListener(this);
 		var cancel = this.cancel.bindAsEventListener(this);
 		var launchAirplaneModeHelp = this.launchAirplaneModeHelp.bindAsEventListener(this);
 		this.flightMode = this.barAssistant.getAirplaneMode();
@@ -16,7 +18,6 @@
 		else
 			this.controller.get('flightmode').innerHTML = "<div class='info-icon' id='airplaneInfo' x-mojo-touch-feedback='immediate'></div>" + $L('Airplane Mode');
 			
-		this.controller.get('poweroff').addEventListener(Mojo.Event.tap, powerOff);
 		this.controller.get('flightmode').addEventListener(Mojo.Event.tap, flightMode);
 		this.controller.get('cancel').addEventListener(Mojo.Event.tap, cancel);
 		if(this.controller.get('airplaneInfo'))
@@ -18,6 +19,9 @@
 			
 		this.controller.get('poweroff').addEventListener(Mojo.Event.tap, powerOff);
 		this.controller.get('flightmode').addEventListener(Mojo.Event.tap, flightMode);
+		this.controller.get('luna').addEventListener(Mojo.Event.tap, luna);
+		this.controller.get('restart').addEventListener(Mojo.Event.tap, restart);
+		this.controller.get('shutdown').addEventListener(Mojo.Event.tap, shutdown);
 		this.controller.get('cancel').addEventListener(Mojo.Event.tap, cancel);
 		if(this.controller.get('airplaneInfo'))
 			this.controller.get('airplaneInfo').addEventListener(Mojo.Event.tap, launchAirplaneModeHelp);
@@ -46,6 +50,22 @@
 		}
 		this.controller.window.close();
 	},
+		
+	java: function() {
+		this.controller.serviceRequest('palm://ca.canucksoftware.systoolsmgr', {
+			method: "javaRestart",
+			parameters: {}
+		});
+		this.controller.window.close();
+	},
+	
+	luna: function() {
+		this.controller.serviceRequest('palm://ca.canucksoftware.systoolsmgr', {
+			method: "lunaRestart",
+			parameters: {}
+		});
+		this.controller.window.close();
+	},
 	
 	flightMode:function(event) {
 		
@@ -47,6 +67,16 @@
 		this.controller.window.close();
 	},
 	
+	restart: function() {
+		this.barAssistant.machineReboot("Restart selected by User in PowerOff Dialog");
+		this.controller.window.close();
+	},
+	
+	shutdown:function() {
+		this.barAssistant.machineOffNow("Shutdown request by User");
+		this.controller.window.close();
+	},
+
 	flightMode:function(event) {
 		
 		var targetId = event.target.id;
